gix/src/repository/object.rs

```diff
@@ -173,11 +175,11 @@ impl crate::Repository {
         constraint: PreviousValue,
     ) -> Result<Reference<'_>, tag::Error> {
         let tag = gix_object::Tag {
-            target: target.as_ref().into(),
+            target: target.into(),
```

Comment:
A very advanced version of `momo` would probably be able to remove `as_ref()` calls by itself one day. For now I find it acceptable to do this by hand though after `momo`ing a method.

Reply:
Resolved by the new internal https://github.com/Byron/gitoxide/pull/992#issuecomment-1697285009